src/JustSaying/Messaging/Middleware/Handle/HandlerInvocationMiddleware.cs

```diff
@@ -5,9 +5,10 @@
 namespace JustSaying.Messaging.Middleware;
 
 /// <summary>
-/// This middleware is responsible for resolving a message handler and calling it.
+/// This middleware is responsible for recalling a previously resolved message handler and calling it.
 /// </summary>
 /// <typeparam name="T">The type of the message that the message handler handles.</typeparam>
+[Obsolete("This class is obsolete and will be removed in a future version. Please use HandlerResolveInvocationMiddleware instead.")]
```

Comment:
I would prefer to avoid doing this, as otherwise it's a build-breaking change for people using `TreatWarningsAsErrors=true`.

This feels more like a _differnent_ way to achieve something, rather than a _wrong_ way.

We could obsolete it later if we wanted to do a full-on replace.